TTCN3_Project/javasrc/CANPortFilter/CANPortFilter.java

```diff
@@ -0,0 +1,84 @@
+package CANPortFilter;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+
+import org.etsi.ttcn.tri.TriAddress;
+import org.etsi.ttcn.tri.TriComponentId;
+import org.etsi.ttcn.tri.TriMessage;
+import org.etsi.ttcn.tri.TriPortId;
+import org.etsi.ttcn.tri.TriStatus;
+
+import com.testingtech.ttcn.tri.PortFilter;
+
+public class CANPortFilter extends PortFilter {
+	
+	File speedFile = new File("resource/Speed");
+	BufferedReader br = null;
+	String speed = null;
```

Comment:
For global variables, a scope should be defined. In addition, global variables should be initialized in the constructor.